node/core/pvf/common/src/worker/security.rs

```diff
@@ -186,3 +186,98 @@ pub mod landlock {
 		}
 	}
 }
+
+// TODO: Add a check for whether seccomp is supported and warn if not, like we do for landlock.
+//       Ideally we would have <https://github.com/rust-vmm/seccompiler/issues/51>.
+//       In lieu of that we can try restricting a dummy thread like we do for landlock.
+/// Process sandboxing using seccomp, a Linux facility for blocking syscall execution.
+///
+/// Currently we only block two syscalls, `socket` and `socketpair`, to effectively disable
+/// networking.
+///
+/// # Action on Caught Syscall
+///
+/// TODO: Update with the action and explain why given action was chosen.
+#[cfg(target_os = "linux")]
+pub mod seccomp {
+	use seccompiler::*;
+	use std::collections::BTreeMap;
+
+	/// The action to take on caught syscalls.
+	#[cfg(not(test))]
+	const CAUGHT_ACTION: SeccompAction = SeccompAction::KillProcess;
+	/// Don't kill the process when testing...
+	#[cfg(test)]
+	const CAUGHT_ACTION: SeccompAction = SeccompAction::Errno(libc::EACCES as u32);
+
+	#[derive(thiserror::Error, Debug)]
+	pub enum Error {
+		#[error(transparent)]
+		Seccomp(#[from] seccompiler::Error),
+
+		#[error(transparent)]
+		Backend(#[from] seccompiler::BackendError),
+	}
+
+	pub type Result<T> = std::result::Result<T, Error>;
+
+	// TODO: Compile the filter at build-time rather than runtime.
+	/// Applies a `seccomp` filter to disable networking for the PVF threads.
+	pub fn try_restrict_thread() -> Result<()> {
+		// Build a `seccomp` filter which by default allows all syscalls except those blocked in the
+		// blacklist.
+		let mut blacklisted_rules = BTreeMap::default();
+
+		blacklisted_rules.insert(libc::SYS_socketpair, vec![]);
+		blacklisted_rules.insert(libc::SYS_socket, vec![]);
```

Comment:
Hm... there could still be a way around this *if* there's an existing socket already created; the attacker could then just probably reconfigure/repurpose it. (I've never tried that but I supposed it should work?) Perhaps we could blacklist all socket-related syscalls? `setsockopt`, `getsockopt`, `connect`, `accept`, `listen`, `bind`, `sendto`, `recvfrom`, `sendmsg`, `recvmsg`, `sendmmsg`, `recvmmsg`, and I also think there were some `compat_*` variant of some of these.

Reply:
The worker thread communicates with the host through a unix socket, I think restricting `send`/`recv` would break that communication?

Reply:
How is socket-repurposing done? Should be impossible to use them for networking if we block everything but local sockets in the main thread (like birdcage does). Then we don't need to let through all the other syscalls.